### ISSUE TITLE
Legend style should match line style

### ISSUE THREAD

**Issue by brylie** [grafana/grafana#13594]
Grafana version: 5.2.4

When displaying legend on a line chart, the legend style should match the line style. This includes dash pattern.

Currently, only a solid line is shown, making it unnecessarily difficult to read the chart:

![screenshot_20181010_135808](https://user-images.githubusercontent.com/17307/46731972-d1b00500-cc94-11e8-9075-f38318f19ca0.png)

**Comment by paucutrina** [comment]
Indeed. I believe that it is something essential. It would be also interesting to add the possibility to select multiple types of markers (square, triangle...) and choose to fill them or not.

**Comment by yj0207** [comment]
Agreed. This is a much-needed feature.

**Comment by hmmcclish** [comment]
I would love to see this feature, among other reasons, so that I can make my home humidity graph use threshold colors for both my indoor sensor series and my outdoor 3rd party data, but use a dashed line for the 3rd party data to distinguish it from the data I'm gathering. Otherwise the graph is too hard to read when both values are within the same threshold:

![image](https://user-images.githubusercontent.com/3462067/181376236-a79e5045-5185-4bc4-9679-20414a788e6f.png)

I would like to add that this feature and similar features were much requested when I was a Power BI junior data analyst in a previous life.

**Comment by patrikrask** [comment]
Agree. Would be very good to have!

**Comment by wzamazon** [comment]
Agree. Would be good to have!

**Comment by arndl** [comment]
Fully agree. I have lines that are grouped by color and differenciated by line style. But it cannot be seen in legend.

**Comment by haeraeus** [comment]
Still a desirable feature that I expected to be standard

**Comment by ichderarnd** [comment]
Please don‘t close this issue since legends are very confusing when different line styles were used with the same color.

**Comment by mbtx2** [comment]
This is an important feature.

**Comment by seidnerj** [comment]
This is critical for color blind people, pls implement this 🙏

**Comment by leeoniya** [comment]
was fixed by https://github.com/grafana/grafana/pull/87558, in 11.1

**Comment by ichderarnd** [comment]
Wow, thanks a lot! 👍🏻
